components/connector-service/internal/certificates/headerparser.go

```diff
@@ -0,0 +1,126 @@
+package certificates
+
+import (
+	"github.com/kyma-project/kyma/components/connector-service/internal/apperrors"
+	"net/http"
+	"regexp"
+)
+
+const ClientCertHeader = "X-Forwarded-Client-Cert"
+
+type HeaderParser interface {
+	ParseCertificateHeader(r http.Request) (CertInfo, apperrors.AppError)
+}
+
+type headerParser struct {
+	country      string
+	locality     string
+	organization string
+	unit         string
+	province     string
+	central      bool
+}
+
+type CertInfo struct {
+	Hash    string
+	Subject string
+}
+
+func NewHeaderParser(country, province, locality, organization, unit string, central bool) HeaderParser {
+	return &headerParser{
+		country:      country,
+		locality:     locality,
+		organization: organization,
+		unit:         unit,
+		province:     province,
+		central:      central,
+	}
+}
+
+func (hp *headerParser) ParseCertificateHeader(r http.Request) (CertInfo, apperrors.AppError) {
+	certHeader := r.Header.Get(ClientCertHeader)
+
+	if certHeader == "" {
+		return CertInfo{}, apperrors.BadRequest("Certificate header is empty")
+	}
+
+	subjectRegex := regexp.MustCompile(`Subject="(.*?)"`)
+
+	subjects := extractFromHeader(certHeader, subjectRegex)
+
+	hashRegex := regexp.MustCompile(`Hash=([0-9a-f]*)`)
+
+	hashes := extractFromHeader(certHeader, hashRegex)
+
+	certInfos := createCertInfos(subjects, hashes)
+
+	if hp.central {
```

Comment:
I would suggest making a subject verification function as a `headerParser` field and assign it in the constructor. Therefore we would use the `central` value only in the constructor.